api/src/main/java/org/apache/iceberg/expressions/Expressions.java

```diff
@@ -105,6 +105,16 @@ public static Expression not(Expression child) {
     return new UnboundPredicate<>(Expression.Operation.NOT_EQ, ref(name), value);
   }
 
+  public static <T> UnboundPredicate<T> in(String name, T value, T... values) {
+    Preconditions.checkNotNull(value, "in predicate must include at least one value");
```

Comment:
All of the values should be non-null. Null values can only be selected with `isNull`. To select from a set or a null value, the predicate must be `or(isNull(t), in(t, 1, 2, 3))`

Reply:
Got it and will add the check for it.